scheduler/reconcile.go

```diff
@@ -961,6 +992,99 @@ func (a *allocReconciler) computeStop(group *structs.TaskGroup, nameIndex *alloc
 	return stop
 }
 
+// computeStopByReconnecting helps build the stop set by analyzing the reconnecting
+// and untainted sets. The function accepts and decrements a target number of
+// allocations to remove. If at any time that reaches zero, the function exits.
+// If there are still allocations to be removed after analyzing the full reconnecting
+// set, that count is returned.
+// The following rules are enforced.
+// * If the reconnecting alloc's desired status in not run, stop it.
+// * If the user-specified desired transition for the reconnecting alloc is not run, stop it.
+// * If the reconnecting alloc has been rescheduled, compare the node NormScores and keep the best.
```

Comment:
I have 100% been mentally modeling this all wrong. I've been optimizing function comments for code maintainers rather than doc strings. 

Updated in next push.